### PR TITLE
Updated the task documentation for 4.* version @vincentdass @kmkumaran 

### PR DESCRIPTION
**Pull request by azooinmyluggage**
Updated the task documentation for 4.* version.
@vincentdass @kmkumaran please take a look.

### CODE REVIEW

File: Tasks/AzureRmWebAppDeploymentV4/README.md [hunk 1]
```diff
@@ -18,10 +18,12 @@ Please report a problem at [Developer Community Forum](https://developercommunit
 
 The following pre-requisites need to be setup in the target machine(s) for the task to work properly.
 
-##### Azure Web App
+##### Azure Web App or Azure Function
 
 The task is used to deploy a Web  project to an existing Azure Web App. The Web App should exist prior to running the task. The Web App can be created from the [Azure portal](https://azure.microsoft.com/en-in/documentation/videos/azure-app-service-web-apps-with-yochay-kiriaty/) and [configured](https://azure.microsoft.com/en-us/documentation/articles/web-sites-configure/) there. Alternatively, the [Azure PowerShell task](https://github.com/Microsoft/vsts-tasks/tree/master/Tasks/AzurePowerShell) can be used to run [AzureRM PowerShell scripts](https://msdn.microsoft.com/en-us/library/mt619237.aspx) to provision and configure the Web App.
 
+The task can also be used to deply [Azure Functions] (https://azure.microsoft.com/en-in/services/functions/) (Windows/Linux/Containers).
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeploymentV4/README.md]: image
please remove spacing before (

File: Tasks/AzureRmWebAppDeploymentV4/README.md [hunk 2]
```diff
@@ -2,11 +2,11 @@
...
+The Azure App Service Deployment task is used to update different Azure App Service to deploy [Web Apps](https://azure.microsoft.com/en-in/documentation/articles/app-service-web-overview/), [Functions](https://docs.microsoft.com/en-us/azure/azure-functions/) and [WebJobs](https://azure.microsoft.com/en-us/blog/webjobs-goes-into-full-production/) to Azure. The task works on cross platform VSTS agents running Windows, Linux or Mac and uses the underlying deployment technologies of [Web Deploy](https://www.iis.net/downloads/microsoft/web-deploy), RunFromZip, Zip Deploy, Containers and [Kudu REST APIs](https://github.com/projectkudu/kudu/wiki/REST-API).
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeploymentV4/README.md]: Should we include publish profile too?

File: Tasks/AzureRmWebAppDeploymentV4/README.md [hunk 3]
```diff
@@ -2,11 +2,11 @@
...
 
-The task works for [ASP.NET](https://www.visualstudio.com/en-us/docs/release/examples/azure/azure-web-apps-from-build-and-release-hubs), [ASP.NET Core](https://www.visualstudio.com/en-us/docs/release/examples/azure/aspnet-core10-azure-web-apps), PHP, Java, Python and [Node.js](https://www.visualstudio.com/en-us/docs/release/examples/nodejs/node-to-azure-webapps) based web applications.
+The task works for [ASP.NET](https://www.visualstudio.com/en-us/docs/release/examples/azure/azure-web-apps-from-build-and-release-hubs), [ASP.NET Core](https://www.visualstudio.com/en-us/docs/release/examples/azure/aspnet-core10-azure-web-apps), PHP, Java, Python, Go and [Node.js](https://www.visualstudio.com/en-us/docs/release/examples/nodejs/node-to-azure-webapps) based web applications.
 
-The task can be used to deploy different Azure App Services like Function App, Web App on Windows, Web App on Linux, Web App for Containers and Azure App Service Environments.
+The task can be used to deploy different Azure App Services like Function App (Windows/Linux/Containers), Web App on Windows, Web App on Linux, Web App for Containers and Azure App Service Environments.
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeploymentV4/README.md]: and Apps configured under App Service Environments

File: Tasks/AzureRmWebAppDeploymentV4/README.md [hunk 4]
```diff
@@ -30,20 +32,31 @@ Create the [ARM](https://azure.microsoft.com/en-in/documentation/articles/resour
...
 * *Container Registry* 
 
-The task defaults to Web Deploy technology on a Windows Agent when the target is Web App for Windows. On other platforms (for any App service type), the task relies on [Kudu REST APIs](https://github.com/projectkudu/kudu/wiki/REST-API) to deploy the Web App.
+* *Zip Deploy* 
+
+* *RunFromZip* 
+
+By default the task tries to select the appropriate deployment technology given the input package, app service type and agent OS.
+
+* For msdeploy package, use Web Deploy
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeploymentV4/README.md]: msdeploy (MSBuild generated package)

File: Tasks/AzureRmWebAppDeploymentV4/README.md [hunk 5]
```diff
@@ -30,20 +32,31 @@ Create the [ARM](https://azure.microsoft.com/en-in/documentation/articles/resour
...
-The task needs the Azure PowerShell version to be installed on the automation agent, and that can be done easily using the [Azure PowerShell Installer v1.3.0](https://github.com/Azure/azure-powershell/releases/tag/v1.3.0-March2016).
-
 ## Deployment
 
+Several deployment methods are available in this task. Web Deploy (msdeploy.exe) is the default option. To change the deployment option, expand Additional Deployment Options and enable Select deployment method to choose from additional package-based deployment options.
+
 Based on the type of Azure App Service and VSTS agent, the task chooses a suitable deployment technology. The different deployment technologies used by the task are:
 * *Web Deploy* 
 
 * *Kudu REST APIs*
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeploymentV4/README.md]: Kudu Zip API

File: Tasks/AzureRmWebAppDeploymentV4/README.md [hunk 6]
```diff
@@ -85,7 +103,9 @@ The task is used to deploy a Web  project to an existing Azure Web App. The mand
 
 * **Package or Folder\*:** Location of the Web App zip package or folder on the automation agent or on a UNC path accessible to the automation agent like, \\\\BudgetIT\\Web\\Deploy\\Fabrikam.zip. Predefined system variables and wild cards like, $(System.DefaultWorkingDirectory)\\\***.zip can be also used here.
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeploymentV4/README.md]: $(System.DefaultWorkingDirectory)/**/*.zip

File: Tasks/AzureRmWebAppDeploymentV4/README.md (hunk 1 above)
Comment (azooinmyluggage) [reply Tasks/AzureRmWebAppDeploymentV4/README.md]: Done

File: Tasks/AzureRmWebAppDeploymentV4/README.md (hunk 3 above)
Comment (azooinmyluggage) [reply Tasks/AzureRmWebAppDeploymentV4/README.md]: Done

File: Tasks/AzureRmWebAppDeploymentV4/README.md (hunk 2 above)
Comment (azooinmyluggage) [reply Tasks/AzureRmWebAppDeploymentV4/README.md]: I will add a separate pr

File: Tasks/AzureRmWebAppDeploymentV4/README.md (hunk 4 above)
Comment (azooinmyluggage) [reply Tasks/AzureRmWebAppDeploymentV4/README.md]: done

**Pull request merged**